### DISCUSSION TITLE
How to use nchat on Android with Termux?

### DISCUSSION THREAD

**Discussion by John4650-hub** [d99kris/nchat#284] [Q&A]
Is it possible to install this tool on android with termux?

**Comment by d99kris** [comment]
Hi @John4650-hub - good question! I just committed a [fix](https://github.com/d99kris/nchat/commit/89954a2) for building nchat on Termux. It works with Telegram, but WhatsApp is not supported currently (do let me know if this would be useful, I can take another stab at it, but it seems it's not straight-forward).

To build and install nchat in Termux I did this:
```
pkg install cmake clang golang ccache gperf file readline libsqlite openssl
git clone https://github.com/d99kris/nchat && cd nchat
mkdir build && cd build
cmake -DCMAKE_INSTALL_PREFIX=${PREFIX} -DHAS_DYNAMICLOAD=OFF -DHAS_WHATSAPP=OFF ..
make -sj8
make install
```
Screenshot of nchat in Termux on my Samsung Galaxy S22:
![nchat-termux-screenshot](https://github.com/user-attachments/assets/4cee975b-9e44-4fe0-a2e4-6a2ff4d64634)

**Reply by John4650-hub** [reply]
Thank you
It's absolutely useful. Personally, i would also prefer using whatsapp in my terminal rather than installing the application from play store.

**Reply by d99kris** [reply]
Hi @John4650-hub - support for building nchat with WhatsApp support under Termux has been added in 71fc941.

To build (with Telegram and WhatsApp support) under Termux, one must set certain build flags (HAS_DYNAMICLOAD=OFF and HAS_STATICGOLIB=OFF). The following commands can be used:
```
pkg install cmake clang golang ccache gperf file readline libsqlite openssl
git clone https://github.com/d99kris/nchat && cd nchat
mkdir build && cd build
cmake -DCMAKE_INSTALL_PREFIX=${PREFIX} -DHAS_DYNAMICLOAD=OFF -DHAS_STATICGOLIB=OFF ..
make -sj8
make install
```

Also note: For setting up a WhatsApp profile in nchat under Termux running on a phone, it will be easiest to ssh into the phone, so that the authentication Qr code can be viewed on a different device, and scanned using the phone in the WhatsApp app.

Screenshot of nchat with a WhatsApp profile in Termux on my S22:
![screenshot-termux-whatsapp](https://github.com/user-attachments/assets/7e581651-ece3-4c88-9ddc-d33792f4a248)

**Comment by d99kris** [comment]
**Update as of 2025 Oct 12th:**

The `make.sh` script has been extended to support Termux now, so it should be possible to just:
```
git clone https://github.com/d99kris/nchat && cd nchat
./make.sh deps
./make.sh build && ./make.sh install
```

A few bugs related to running external commands from nchat (external editor, spell check, etc) on Termux has also been fixed.

**Reply by John4650-hub** [reply]
This is alot better. Thank you
